### PR TITLE
sqlite: remove some allocs

### PR DESCRIPTION
**Pull request by bradfitz**

### CODE REVIEW

File: sqlite_test.go [hunk 1]
```diff
@@ -835,6 +836,54 @@ func TestSetWALHook(t *testing.T) {
...
+	if err != nil {
+		t.Fatal(err)
+	}
+	n := testing.AllocsPerRun(10000, func() {
+		if _, err := stmt.Exec(); err != nil {
+			t.Fatal(err)
+		}
+	})
+	const maxAllocs = 3 // as of Go 1.20
+	if n > maxAllocs {
```
Comment (maisem) [inline sqlite_test.go:855]: What was it before?

**Review by maisem — APPROVED** [review]

File: sqlite_test.go (hunk 1 above)
Comment (bradfitz) [reply sqlite_test.go:855]: 5 IIRC?

**Comment by bradfitz** [comment]
@crawshaw, you fine with this bumping minimum required Go version to 1.20?

Feel free to merge if happy.

**Review by crawshaw — APPROVED** [review]
looks right!

**Pull request merged**